### PR TITLE
py-xarray: add v2023.7.0

### PR DESCRIPTION
**Pull request by genric**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-xarray/package.py [hunk 1]
```diff
@@ -51,22 +52,26 @@ class PyXarray(PythonPackage):
     depends_on("python@3.6:", when="@0.14:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xarray/package.py:53]: setuptools-scm@7: is required now

File: var/spack/repos/builtin/packages/py-xarray/package.py [hunk 2]
```diff
@@ -75,8 +80,8 @@ class PyXarray(PythonPackage):
     depends_on("py-zarr", when="+io", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xarray/package.py:81]: pydap only `when="^python@:3.9"`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-xarray/package.py (hunk 1 above)
Comment (genric) [reply var/spack/repos/builtin/packages/py-xarray/package.py:53]: done

File: var/spack/repos/builtin/packages/py-xarray/package.py (hunk 2 above)
Comment (genric) [reply var/spack/repos/builtin/packages/py-xarray/package.py:81]: done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**